commons/src/main/java/io/aiven/kafka/tieredstorage/commons/UniversalRemoteStorageManagerConfig.java

```diff
@@ -157,6 +189,22 @@ ObjectStorageFactory objectStorageFactory() {
         return objectFactory;
     }
 
+    Long segmentManifestCacheSize() {
```

Comment:
Do we really need to make it `Long` and not primitive `long`? I guess it's never null here because of config definition and moreover there is an unboxing on the next line. As I suggested in another comment I would get rid of null-ability here using `long` and returning raw value and check for -1 in the other place.

Reply:
Will be addressed later in another PR as mentioned in another comment.